### ISSUE TITLE
`mnemonicToEntropy` not returning correct entropy

### ISSUE THREAD

**Issue by cgcardona** [bitcoinjs/bip39#70]
I found a call to `mnemonicToEntropy` which isn't returning the correct entropy. To reproduce:

```js
let entropy = "ea9d80e70b08bbfb6b7f2bca56c88ddb2c2d003e675a855779d81e11f3009346a";
let mnemonic = bip39.entropyToMnemonic(entropy);
// 'tunnel ugly degree bid merry world pupil tornado ski rent casino rent security length wet twice luxury rookie invite destroy busy leopard escape shoe'

let rand = bip39.mnemonicToEntropy(mnemonic);
// 'ea9d80e70b08bbfb6b7f2bca56c88ddb2c2d003e675a855779d81e11f3009346'
```

The returned entropy is missing an `a` at the end. This is the only case where I've found that `mnemonicToEntropy` doesn't return the correct entropy.

Thanks for a great library!

**Comment by dcousens** [comment]
Interesting,  but,  this actually isn't our fault!

``` js
let buffer = Buffer.from('ea9d80e70b08bbfb6b7f2bca56c88ddb2c2d003e675a855779d81e11f3009346a', 'hex')
// <Buffer ea 9d 80 e7 0b 08 bb fb 6b 7f 2b ca 56 c8 8d db 2c 2d 00 3e 67 5a 85 57 79 d8 1e 11 f3 00 93 46>
```

You passed a 65-byte hex string,  and a 32-byte Buffer is parsed.
Why does Node not throw?  Ask them,  but,   alas.

**Comment by dcousens** [comment]
Keeping open for now,  as I'd like to hear your opinion,  this is certainly a confusing result.

**Comment by cgcardona** [comment]
Looking back through my console I see the issue. TLDR adding the `a` on the end was a typo in my test suite.

Both the 64 and 65 char entropy strings produce the same mnemonic 

```js
let a = bip39.entropyToMnemonic('ea9d80e70b08bbfb6b7f2bca56c88ddb2c2d003e675a855779d81e11f3009346')
// 'tunnel ugly degree bid merry world pupil tornado ski rent casino rent security length wet twice luxury rookie invite destroy busy leopard escape shoe'

let b = bip39.entropyToMnemonic('ea9d80e70b08bbfb6b7f2bca56c88ddb2c2d003e675a855779d81e11f3009346a')
// 'tunnel ugly degree bid merry world pupil tornado ski rent casino rent security length wet twice luxury rookie invite destroy busy leopard escape shoe'
```

Node's `Crypto.randomBytes` correctly returned a 64 character hex encoded string  but when I manually copied the 64 char string to my fixtures JSON I added an `a` due to my vim workflow. 

So  my `entropyToMnemonic` tests passed but the `mnemonicToEntropy` tests failed because my assertion was for the 65 character string. 

Apologies for the false flag. Again, thanks for a great lib. 🎩

**Comment by lp12ucw** [comment]
What’s this ?

Sent from my iPhone

On 16 Mar 2018, at 11:03, Daniel Cousens <notifications@github.com> wrote:

Closed #70.
